common/types/string.go

```diff
@@ -497,3 +523,225 @@ func parseClausePrecision(lastStrIndex int, formatStr string) (int, int, error)
 func isASCIIDigit(r rune) bool {
 	return r <= unicode.MaxASCII && unicode.IsDigit(r)
 }
+
+func matchLanguage(locale string) language.Tag {
+	tag, _ := language.MatchStrings(langMatcher, locale)
+	return tag
+}
```

Comment:
Do we need this function? If we omitted it and instead required callers provide a language tag, could we avoid having to maintain the `initTags()` implementation?

Reply:
The [definition](https://github.com/golang/text/blob/ec5565b1b747ce5ca569aeefc09e737b479a12ac/internal/language/language.go#L35) of `language.Tag` is fairly intricate, so I'm not sure it'd be good to leave supplying it up to expression writers.

As far as using `message.MatchLanguage` directly goes (intuitively it'd work instead of `initTags`, and was the first thing I tried, unfortunately): If the Go standard library's [default catalog wasn't empty by default](https://github.com/golang/text/blob/e328d63cff14134669501e0e154e4f141c784322/message/catalog.go#L26), then I don't think `initTags` would be necessary, but trying to use the default catalog as-is results in [no locale support](https://go.dev/play/p/CyqWfmrG7U1) (the number should use a comma decimal separator given the locale).

As far as I can tell this is the intended behavior; I think the intention is for programs to explicitly add the languages they support.

Reply:
> As far as I can tell this is the intended behavior; I think the intention is for programs to explicitly add the languages they support.

This is my exact point. Let the application programmer. In the trivial case, I would expect the application programmer to pass in `language.MustParse("locale I care about")`. If the application programmer has a localized app, it is likely they have already performed a language match and have the appropriate tag.

Reply:
How about we make a language tag list with only one member (the user-given language tag), so that if we have a CEL expression:

```
"%f".format([3.14], "fr_FR")
```

We then make a tag list containing only the French locale, and then match against that? I don't think there's much overhead in building this list every time. If an invalid locale is passed, the default behavior should be in line with `en_US` (or if it isn't, we can build a two-member list every time with `en_US` as the fallback option).

(Sorry if that's what you were suggesting - I had interpreted it to read as passing those tags in on the Golang side.)

Reply:
For the case of a `format(..., locale)` would you even need a tag list? Couldn't you just call `language.Parse(locale)?

Also, I'm skeptical of the value of a `format(..., locale)` function. I expect the vast majority of users to want to call format without locale and instead configure the `Env` with the locale.

Reply:
Right - I'll remove the `...locale` overload and add it into the `env` and mark this as resolved when I do so.

Reply:
(marking as resolved with the addition of `LocaleOptionBinding`)